modular_skyrat/code/modules/projectiles/guns/energy/kinetic_accelerator.dm

```diff
@@ -8,6 +8,8 @@
 			to_chat(loc, "<span class='warning'>[src] silently charges up.</span>")
 		update_icon()
 		overheat = FALSE
+	else //this is a terrible solution, but it ensures that it wont be stuck on dischaged if it fails to reload in an obj
+		addtimer(CALLBACK(src, .proc/reload), overheat_time * 2, TIMER_STOPPABLE)
```

Comment:
I think you're going to need a deltimer() here so this doesn't just keep creating more and more timers.

Reply:
Probably a good idea, yeah, even though i think it'd be hard to create stacking on this specific situation.